### PR TITLE
Add SPV_EXT_optnone extension

### PR DESCRIPTION
**Pull request by MrSidims**

### CODE REVIEW

**Review by kpet — COMMENTED** [review]
We would be interested in seeing this extension published as an EXT instead.

File: README.md [hunk 1]
```diff
@@ -74,6 +74,7 @@ Khronos SPIR-V Registry](https://www.khronos.org/registry/spir-v/).
 * [SPV_EXT_shader_tile_image               ]( http://htmlpreview.github.io/?https://github.com/KhronosGroup/SPIRV-Registry/blob/main/extensions/EXT/SPV_EXT_shader_tile_image.html)
 * [SPV_EXT_shader_viewport_index_layer     ]( http://htmlpreview.github.io/?https://github.com/KhronosGroup/SPIRV-Registry/blob/main/extensions/EXT/SPV_EXT_shader_viewport_index_layer.html)
 * [SPV_EXT_ycbcr_attachments               ]( http://htmlpreview.github.io/?https://github.com/KhronosGroup/SPIRV-Registry/blob/main/extensions/EXT/SPV_EXT_ycbcr_attachments.html)
+* [SPV_EXT_optnone                  __     ]( http://htmlpreview.github.io/?https://github.com/KhronosGroup/SPIRV-Registry/blob/main/extensions/EXT/SPV_EXT_optnone.html)
```
Comment (bashbaug) [inline README.md:77]: Could you please keep this list alphabetized?  Thanks!

**Review by bashbaug — APPROVED** [review]

File: extensions/EXT/SPV_EXT_optnone.asciidoc [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+[cols="1,4,4",options="header",width="100%"]
+|====
+2+| Function Control | <<Capability,Enabling Capabilities>> |
+0x10000 | *OptNoneEXT* +
+Strong request, to the extent possible, to not optimize the function. Only functions
+with *Inline* _Function Control_ mask can be considered to be inlined into the
+function. +
+This function should never be inlined. +
+It must not be used together with *Inline* bit. +
+| *CapabilityOptNoneEXT*
```
Comment (kpet) [inline extensions/EXT/SPV_EXT_optnone.asciidoc:101]:
```suggestion
| *OptNoneEXT*
```

File: extensions/EXT/SPV_EXT_optnone.asciidoc [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+~~~~~~~~~~~~
+
+Modify Section 3.31, "Capability", adding these rows to the Capability table:
+
+--
+[options="header"]
+|====
+2+^| Capability ^| Implicitly Declares
+| 6094 | *OptNoneEXT* +
+Allow to use *OptNoneEXT* Function Control mask value |
```
Comment (bashbaug) [inline extensions/EXT/SPV_EXT_optnone.asciidoc:82]: Confirming: there are no concerns using **OptNoneEXT** as both the name of the Capability and the name of the Function Control bit?

File: extensions/EXT/SPV_EXT_optnone.html [hunk 4]
```diff
@@ -0,0 +1,706 @@
+<!DOCTYPE html>
+<html lang="en">
+<head>
+<meta charset="UTF-8">
+<meta http-equiv="X-UA-Compatible" content="IE=edge">
+<meta name="viewport" content="width=device-width, initial-scale=1.0">
+<meta name="generator" content="Asciidoctor 2.0.16">
```
Comment (kpet) [inline extensions/EXT/SPV_EXT_optnone.html:7]: Looks like this was not generated with the version of asciidoctor provided in the docker image. We're working towards automatically generating and publishing specs (see https://github.com/KhronosGroup/SPIRV-Registry/issues/270) and one of the necessary steps involves making sure that all specs can be generated with the version provided in the docker image. It seems that we have not updated EXT extensions yet and this spec looks rather straightforward so I think re-generating the HTML needn't be a blocker to this PR going in (but would be nice to do if you can).

**Review by kpet — APPROVED** [review]

**Comment by bashbaug** [comment]
Merging as discussed in the September 4th teleconference.

**Pull request merged**